src/include/gpopt/translate/CTranslatorQueryToDXL.h

```diff
@@ -138,6 +138,10 @@ class CTranslatorQueryToDXL
 	// CTE producer IDs defined at the current query level
 	UlongBoolHashMap *m_cteid_at_current_query_level_map;
 
+	// id of current query (and for nested queries), used for 
+	// deduplication of target (result) entries (unique index)
```

Comment:
... it's used for correct assigning of relation links to target relation of DML query